src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -1935,10 +1935,19 @@ private static bool IsUserDefinedTrueOrFalse(BoundUnaryOperator @operator)
                 else if ((highestBoundExpr as BoundConversion)?.Conversion.IsTupleLiteralConversion == true)
                 {
                     var tupleLiteralConversion = (BoundConversion)highestBoundExpr;
-                    var convertedTuple = (BoundConvertedTupleLiteral)tupleLiteralConversion.Operand;
-                    type = convertedTuple.NaturalTypeOpt;
-                    convertedType = tupleLiteralConversion.Type;
-                    conversion = tupleLiteralConversion.Conversion;
+                    if (tupleLiteralConversion.Operand.Kind == BoundKind.ConvertedTupleLiteral)
+                    {
+                        var convertedTuple = (BoundConvertedTupleLiteral)tupleLiteralConversion.Operand;
+                        type = convertedTuple.NaturalTypeOpt;
+                        convertedType = tupleLiteralConversion.Type;
+                        conversion = tupleLiteralConversion.Conversion;
+                    }
+                    else
+                    {
+                        type = tupleLiteralConversion.Operand.Type;
+                        convertedType = tupleLiteralConversion.Type;
+                        conversion = tupleLiteralConversion.Conversion;
```

Comment:
It looks like these two statements are duplicated across both branches. Consider moving them out of the ```if```.